src/components/share/ShareDirective.js

```diff
@@ -69,13 +81,47 @@
                 }).success(function(response) {
                   scope.permalinkValue = response.shorturl;
                   scope.urlShortened = true;
+                  scope.$applyAsync(function() {
+                    // Auto-select the shortened permalink (not on mobiles)
+                    if (!gaBrowserSniffer.mobile) {
+                      $('#permalinkInput')[0].setSelectionRange(0, 9999);
+                    }
+                  });
                 });
               };
 
+              // Use clipboard API to copy URL in OS clipboard
+              scope.copyUrl = function(type) {
+                // Select the permalink anchor text
+                if (type == 'permalink') {
+                  var inputElement = $('#permalinkInput');
+                } else if (type == 'embed') {
+                   var inputElement = $('.ga-embed-input');
+                }
+                inputElement[0].setSelectionRange(0, 9999);
+                var res;
+                // Execute the copy command
+                res = $document[0].execCommand('copy');
+                if (res) {
+                  scope.userMessage = $translate.instant('copy_url_success');
+                } else {
+                  scope.userMessage = $translate.instant('copy_url_fail');
+                }
+
+                // Remove the selections - NOTE: the two following lines
+                // do the job for all three browsers
+                $document[0].getSelection().removeAllRanges();
+                $document[0].getSelection().addRange($document[0]
+                    .createRange());
+              };
+
               // Select the input field on click in order to allow copy/paste
               scope.selectOnClick = function(e) {
```

Comment:
The click event  does the same thing  of the focus event  on l41 ? In which the click event is called and not the focus event ?